dspace-api/src/main/java/org/dspace/app/sherpa/SHERPAService.java

```diff
@@ -103,10 +106,14 @@ public SHERPAResponse searchByJournalISSN(String query)
 
                 HttpEntity responseBody = response.getEntity();
 
-                if (null != responseBody)
-                    sherpaResponse = new SHERPAResponse(responseBody.getContent(), SHERPAResponse.SHERPAFormat.JSON);
-                else
+                if (null != responseBody) {
+                    InputStream content = responseBody.getContent();
+                    sherpaResponse = new SHERPAResponse(content, SHERPAResponse.SHERPAFormat.JSON);
+                    content.close();
```

Comment:
Can we place this content.close() in a finally ? If for whatever reason the parsing fails the inputStream won't be closed. After this change I approve of this PR !

Ps: We should also remember to port this PR to the latest DSpace 7 master.

Reply:
Good point, I did that in my other implementation, I'm not sure why this one was different.
I have rewritten this small section for a try/catch/finally block, and in the same commit I updated the class to conform to DSpace 7 code style for easier porting.